pkg/cmd/secret/shared/shared_test.go

```diff
@@ -0,0 +1,195 @@
+package shared
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestGetSecretEntity(t *testing.T) {
+	tests := []struct {
+		name        string
+		orgName     string
+		envName     string
+		userSecrets bool
+		want        SecretEntity
+		wantErr     bool
+	}{
+		{
+			name:    "org",
+			orgName: "myOrg",
+			want:    Organization,
+		},
+		{
+			name:    "env",
+			envName: "myEnv",
+			want:    Environment,
+		},
+		{
+			name:        "user",
+			userSecrets: true,
+			want:        User,
+		},
+		{
+			name: "defaults to repo",
+			want: Repository,
+		},
+		{
+			name:    "Errors if both org and env are set",
+			orgName: "myOrg",
+			envName: "myEnv",
+			wantErr: true,
+		},
+		{
+			name:        "Errors if both org and user secrets are set",
+			orgName:     "myOrg",
+			userSecrets: true,
+			wantErr:     true,
+		},
+		{
+			name:        "Errors if both env and user secrets are set",
+			envName:     "myEnv",
+			userSecrets: true,
+			wantErr:     true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			entity, err := GetSecretEntity(tt.orgName, tt.envName, tt.userSecrets)
+
+			if tt.wantErr {
+				require.Error(t, err)
```

Comment:
I don't have a strong preference, except for consistency with other tests, but is there a reason why we should be using `require` instead of `assert` in this test file?

Reply:
I switched to `assert` - https://github.com/cli/cli/pull/5134/commits/072e02fe0815c88fd9a3a07ac9bdaabce012ca73

I don't have any strong preferences, just a slight personal preference towards `require` as it behaves similar to other testing libraries I'm used to (stopping test execution when an assertion fails)